src/session/session_api.c

```diff
@@ -636,8 +631,10 @@ __session_open_cursor_int(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *
             WT_RET(__wt_curbackup_open(session, uri, other, cfg, cursorp));
         break;
     case 's':
-        if (WT_PREFIX_MATCH(uri, "statistics:"))
-            WT_RET(__wt_curstat_open(session, uri, other, cfg, cursorp));
+        if (WT_PREFIX_MATCH(uri, "statistics:")) {
+            WT_ASSERT(session, other == NULL);
```

Comment:
I don't quite understand this change, what are we using other for?

Reply:
other is used for additional cursors besides the cursor being opened, and owner. Most types of opens do not need other. I traced all the opens and found out with the removal of join cursors none of the statistics: cursors need other.  The call stack was like this before this change:

SESSION::open_cursor() == __session_open_cursor() -> __session_open_cursor_int(other = statjoin || dupbackup ? to_dup : NULL) -> Dispatch on uri prefix, if statistics: -> __wt_curstat_open(other) -> __wt_curstat_init(other), __wt_cursor_init().

After this change it changed to:
SESSION::open_cursor() == __session_open_cursor() -> __session_open_cursor_int(other = dupbackup ? to_dup : NULL) -> Dispatch on uri prefix, if statistics: -> __wt_curstat_open() -> __wt_curstat_init(other), __wt_cursor_init().

__session_open_cursor_int() is no longer called with other != NULL for a statistics: cursor, only for duplicating a backup cursor. Thus there is no longer a need to pass other to __wt_curstat_open() since it is always NULL.

Reply:
One more thing. join: cursors used other as the cursor that is being joined to in complex joins, I think.

Reply:
Thanks for the explanation!